azure-go-aks/main.go

```diff
@@ -13,6 +13,27 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
 )
 
+func getKubeconfig(ctx *pulumi.Context, cluster *containerservice.ManagedCluster, resourceGroup *resources.ResourceGroup) pulumi.Output {
+	kubeconfig := pulumi.All(cluster.Name, resourceGroup.Name, resourceGroup.ID()).ApplyT(func(args interface{}) (string, error) {
+		clusterName := args.([]interface{})[0].(string)
+		resourceGroupName := args.([]interface{})[1].(string)
+		creds, err := containerservice.ListManagedClusterUserCredentials(ctx, &containerservice.ListManagedClusterUserCredentialsArgs{
```

Comment:
Is there now a ListManagedClusterUserCredentialsOutput function?

The idea is that the code *may* be simplified if you use that since it would let you avoid `pulumi.All` plus unpacking of args. You would pass cluster.Name etc directly to that function.